### PR TITLE
Cleanup handling of `crate_name` for doctests

### PR DESCRIPTION
**Pull request by jyn514**
- Remove unnecessary reallocation
- Remove unnecessary messing around with `queries` for the crate name; it's simpler and faster to go through the TyCtxt instead
- Rename `cratename` -> `crate_name` for consistency with the rest of the compiler

### CODE REVIEW

File: src/librustdoc/doctest.rs [hunk 1]
```diff
@@ -111,19 +113,16 @@ crate fn run(options: Options) -> Result<(), ErrorReported> {
...
-            let crate_name = queries.crate_name()?.peek().to_string();
             let mut global_ctxt = queries.global_ctxt()?.take();
 
             let collector = global_ctxt.enter(|tcx| {
                 let krate = tcx.hir().krate();
                 let crate_attrs = tcx.hir().attrs(CRATE_HIR_ID);
 
-                let mut opts = scrape_test_config(crate_attrs);
-                opts.display_warnings |= options.display_warnings;
+                let opts = scrape_test_config(crate_attrs);
```
Comment (jyn514) [inline src/librustdoc/doctest.rs:122]: Err. I accidentally removed this line.

Comment (CraftSpider) [reply src/librustdoc/doctest.rs:122]: Whoops, my fault for missing that. Honestly, I thought it was part of your changes to name handling at first.

**Comment by jyn514** [comment]
@bors r=CraftSpider

Thanks for the review!

File: src/librustdoc/doctest.rs (hunk 1 above)
Comment (jyn514) [reply src/librustdoc/doctest.rs]: No big deal, I missed it too 😆

**Pull request merged**